Assets/FeatureVisualizeEnemies/EnemyFilterSelector.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyFilterSelector : MonoBehaviour {
+
+    // This script can be attached to enemies who require a special filter to be seen
+    // Attach it and a random filter will be attributed on Awake
+    // The EnemyVisualizer will take care of displaying the right ones
+    public int RequiredFilter;
+
+	void Awake() {
+        RequiredFilter = Random.Range(0, 3);
+        GameObject.Find("EnemyVisualizer").GetComponent<VisualizeEnemies>().updateEnemiesList();
```

Comment:
If the EnemyVisualizer takes care of displaying the right enemies, why does your filter selector have to call explicitly a method from it? Maybe consider using events fired from here and catched by the visualizer (ask @Ezia for help, she'll help you with that).

Reply:
I will take care of that with her help tonight, thanks!